syntax = "proto3";

package reverse_access;

// The reverse access service definition.
service Reverser {
  // Send client resource to server 
  rpc SendResource (ResourceRequest) returns (ResourceReply) {}

  // Subscribe to client to server commands
  rpc SubscribeCommand (CommandRequest) returns (stream CommandReply) {}
}

message CommandRequest {
  string id = 1;
}

message CommandReply {
  string id = 1;
  uint32 command = 2;
  string args = 3;
}

message ResourceRequest {
  string id = 1;
  uint32 command = 2;
  bytes data = 3;
}

message ResourceReply {
  uint32 status = 1;
}

